### PR TITLE
Small fixes for dynamic select

### PR DESCRIPTION
**Pull request by philippjfr**
There was two small issues with select, dynamic overlaying was not specifying which objects it wants to be selecting on and secondly Datasets weren't validating selections applied to them.

### CODE REVIEW

File: holoviews/core/data/__init__.py [hunk 1]
```diff
@@ -316,7 +316,10 @@ def select(self, selection_specs=None, **selection):
         supplied, which will ensure the selection is only applied if the
         specs match the selected object.
         """
-        if selection_specs and not any(self.matches(sp) for sp in selection_specs):
+        selection = {dim: sel for dim, sel in selection.items()
+                     if dim in self.dimensions()+['selection_mask']}
```
Comment (jlstevens) [inline holoviews/core/data/__init__.py]: What is the special string 'selection_mask'? I grepped the code base and it doesn't currently appear anywhere...

**Comment by jlstevens** [comment]
Looks good and the tests are passing. Happy to merge if you say it is ready.

File: holoviews/core/data/__init__.py (hunk 1 above)
Comment (philippjfr) [inline holoviews/core/data/__init__.py]: It's used a bunch in the interfaces and allows applying a boolean mask in ``__getitem__``.

**Comment by philippjfr** [comment]
Ready to merge.

File: holoviews/core/data/__init__.py (hunk 1 above)
Comment (jlstevens) [inline holoviews/core/data/__init__.py]: Ah, it just doesn't seem to appear elsewhere in the codebase as a string like this...

**Pull request merged**